src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,41 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+        LocalDate dateDateFrom = LocalDate.parse(dateFrom, formatter);
+        LocalDate dateDateTo = LocalDate.parse(dateTo, formatter);
+        int[] salaries = new int[names.length];
+        StringBuilder output = new StringBuilder("Report for period ");
+        output.append(dateFrom).append(" - ").append(dateTo).append(System.lineSeparator());
+        for (int i = 0; i < names.length; i++) {
+            for (String d : data) {
+                boolean dateCheck =
+                        (dateDateFrom.isBefore(LocalDate.parse(d.substring(0, 10), formatter))
+                        && dateDateTo.isAfter(LocalDate.parse(d.substring(0, 10), formatter))
+                        || dateDateFrom.isEqual(LocalDate.parse(d.substring(0, 10), formatter))
+                        || dateDateTo.isEqual(LocalDate.parse(d.substring(0, 10), formatter)));
```

Comment:
you call the `substring` method 4 times here to get the same data. Maybe we can move to a separate variable 

Also, you can use `split` method to parse data from string